Add management app documentation (#2929)
wso2/docs-is: en/docs/develop/authenticating-and-authorizing-rest-apis.md

> WSO2 Identity Server supports three ways of API authentication:

Reviewer: ```suggestion
WSO2 Identity Server supports the following API authentication methods:
```

> -   Basic authentication: Uses the user’s credentials in the API invocation

Reviewer: ```suggestion
-   Basic authentication: Use the user’s credentials to invoke the APIs.
```

> -   OAuth 2 common flows: Obtains a token using an oauth2 flow and uses it to invoke the API. The application needs to be created as Management Application. Refer to Service Provider creation [guide](../../learn/adding-and-configuring-a-service-provider/#adding-a-service-provider)

Reviewer: ```suggestion
-   OAuth 2 common flows: Obtains a token using an OAuth2 flow and uses it to invoke the API.
    !!! info
        To invoke the API using OAuth common flows, the application should be a management application. Refer the [service provider creation guide](../../learn/adding-and-configuring-a-service-provider/#adding-a-service-provider) to create a new management application on IS.
```

> -   Client certificate-based: Uses Mutual SSL to authenticate in order to consume the APIs

Reviewer: ```suggestion
-   Client certificate-based: Use mutual SSL to authenticate users in order to invoke the APIs.
```
